File: AtlasWeb/Core/Modules/Tibbi_Surec/Hemodiyaliz_Modulu/HemodialysisOrderFormViewModel.cs
//$77C889E5
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Infrastructure.Helpers;
using System.Collections.Generic;

namespace Core.Controllers
{
    public partial class HemodialysisOrderServiceController
    {
        partial void PreScript_HemodialysisOrderForm(HemodialysisOrderFormViewModel viewModel, HemodialysisOrder hemodialysisOrder, TTObjectContext objectContext)
        {
            Guid? activeEpisodeObjectID = viewModel.GetSelectedEpisodeActionID();
            if (activeEpisodeObjectID.HasValue && hemodialysisOrder.IsOldAction != true)
            {
                EpisodeAction activeEpisodeAction = objectContext.GetObject<EpisodeAction>(activeEpisodeObjectID.Value);

                hemodialysisOrder.SetMandatoryEpisodeActionProperties(activeEpisodeAction, activeEpisodeAction.MasterResource, true);

                hemodialysisOrder.MasterAction = activeEpisodeAction;//?????????????????
                hemodialysisOrder.CurrentStateDefID = HemodialysisOrder.States.Request;
                if (activeEpisodeAction.SubEpisode.StarterEpisodeAction != null && activeEpisodeAction.SubEpisode.StarterEpisodeAction.ProcedureDoctor != null)
                    hemodialysisOrder.ProcedureDoctor = activeEpisodeAction.SubEpisode.StarterEpisodeAction.ProcedureDoctor;
                else if (activeEpisodeAction.ProcedureDoctor != null)
                    hemodialysisOrder.ProcedureDoctor = activeEpisodeAction.ProcedureDoctor;
            }

            List<ResEquipment> resEquipmentList = new List<ResEquipment>();
            var equipmentList = ResEquipment.GetHemodialysisResEquipment(objectContext);
            foreach (var equipment in equipmentList)
            {
                ResEquipment resEquipment = objectContext.GetObject(equipment.ObjectID.Value, equipment.ObjectDefID.Value) as ResEquipment;
                resEquipmentList.Add(resEquipment);
            }
            viewModel.ResEquipments = resEquipmentList.ToArray();
            ContextToViewModel(viewModel, objectContext);
        }

        partial void PostScript_HemodialysisOrderForm(HemodialysisOrderFormViewModel viewModel, HemodialysisOrder hemodialysisOrder, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {
            if (hemodialysisOrder.HemodialysisRequest != null && hemodialysisOrder.IsOldAction != true)
            {
                EpisodeAction activeEpisodeAction = hemodialysisOrder.HemodialysisRequest;

                hemodialysisOrder.SetMandatoryEpisodeActionProperties(activeEpisodeAction, activeEpisodeAction.MasterResource, true);

                hemodialysisOrder.MasterAction = activeEpisodeAction;//?????????????????
                if (hemodialysisOrder.CurrentStateDefID == null)
                {
                    hemodialysisOrder.CurrentStateDefID = HemodialysisOrder.States.Request;
                }

                if (activeEpisodeAction.SubEpisode.StarterEpisodeAction != null && activeEpisodeAction.SubEpisode.StarterEpisodeAction.ProcedureDoctor != null)
                    hemodialysisOrder.ProcedureDoctor = activeEpisodeAction.SubEpisode.StarterEpisodeAction.ProcedureDoctor;
                else if (activeEpisodeAction.ProcedureDoctor != null)
                    hemodialysisOrder.ProcedureDoctor = activeEpisodeAction.ProcedureDoctor;
            }

            if (hemodialysisOrder.SessionCount != null)
            {
                hemodialysisOrder.Amount = hemodialysisOrder.SessionCount;
            }
            else
            {
                hemodialysisOrder.Amount = 0;
            }

            if (hemodialysisOrder.PackageProcedure != null)
            {
                hemodialysisOrder.ProcedureObject = hemodialysisOrder.PackageProcedure;
            }

            //hemodialysisOrder.CurrentStateDefID = HemodialysisOrder.States.Plan;
        }

        partial void AfterContextSaveScript_HemodialysisOrderForm(HemodialysisOrderFormViewModel viewModel, HemodialysisOrder hemodialysisOrder, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {
            hemodialysisOrder.CurrentStateDefID = HemodialysisOrder.States.Plan;
            objectContext.Save();
        }
    }
}

namespace Core.Models
{
    public partial class HemodialysisOrderFormViewModel : BaseViewModel
    {
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Hemodiyaliz_Modulu/HemodialysisOrderFormViewModel.ts
//$5176D4E5
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { HemodialysisOrder } from "NebulaClient/Model/AtlasClientModel";
import { HemodialysisRequest } from "NebulaClient/Model/AtlasClientModel";
import { ResEquipment } from "NebulaClient/Model/AtlasClientModel";
import { ResTreatmentDiagnosisUnit } from "NebulaClient/Model/AtlasClientModel";
import { PackageProcedureDefinition } from "NebulaClient/Model/AtlasClientModel";
import { SKRSDiyalizeGirmeSikligi } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';

export class HemodialysisOrderFormViewModel extends BaseViewModel {
    @Type(() => HemodialysisOrder)
    public _HemodialysisOrder: HemodialysisOrder = new HemodialysisOrder();
    @Type(() => HemodialysisRequest)
    public HemodialysisRequests: Array<HemodialysisRequest> = new Array<HemodialysisRequest>();
    @Type(() => ResEquipment)
    public ResEquipments: Array<ResEquipment> = new Array<ResEquipment>();
    @Type(() => ResTreatmentDiagnosisUnit)
    public ResTreatmentDiagnosisUnits: Array<ResTreatmentDiagnosisUnit> = new Array<ResTreatmentDiagnosisUnit>();
    @Type(() => PackageProcedureDefinition)
    public PackageProcedureDefinitions: Array<PackageProcedureDefinition> = new Array<PackageProcedureDefinition>();
    @Type(() => SKRSDiyalizeGirmeSikligi)
    public SKRSDiyalizeGirmeSikligis: Array<SKRSDiyalizeGirmeSikligi> = new Array<SKRSDiyalizeGirmeSikligi>();
}
